components/datetime/src/date.rs

```diff
@@ -163,6 +165,40 @@ fn week_of_year<T: DateInput>(
     Ok(WeekOfYear(u32::from(week.week)))
 }
 
+/// Returns the week of month according to a calendar with min_week_days = 1.
+///
+/// This is different from what the UTS35 spec describes but the latter is
+/// missing a month of week-of-month field so following the spec would result
+/// in inconsistencies (e.g. in the ISO calendar 2021-01-01 is the last week
+/// of December but 'MMMMW' would have it formatted as 'week 5 of January').
+fn week_of_month<T: DateInput>(
+    datetime: &T,
+    first_weekday: IsoWeekday,
+) -> Result<WeekOfMonth, DateTimeError> {
+    // This value isn't used by week_of() given that we use a calendar with min_week_days = 1.
+    const UNUSED_DAYS_IN_MONTH: u16 = 60;
```

Comment:
> Thought

This seems a little bit strange to me. But I'm not familiar enough with this code to know what alternatives would be. 

As someone who is not familiar with the `week_of` code, I'm not sure why the `min_week_days = 1` is relevant, or why this value is 60. 

> Question

Is the 60 an arbitrary value, since this is "unused," or does this value have to be 60?
> Question

I wonder if we can't do something a bit cleaner using `Option`. 

Could `week_of::week_of` potentially take an `Option`?

Reply:
My bad, I was leaking arithmetic.rs's implementation details there. I've added a simple_week_of() method to arithmetic.rs that encapsulates those details instead. If that's still non-ideal I can add an implementation for t simple_week_of() rather than call into the more complete week_of(); I'd rather avoid having Options complicate the week_of signature...

Reply:
With `min_week_days` = 1 `week_of()` boils down to ceil((day - 1 + (first_day_of_month - first_weekday)) / 7). `num_days_in_previous_unit` & `num_days_in_unit` only come into play for `min_week_days` > 1 in which case a day in the current month/year can get counted as part of the previous or next month/year.